docs/extras/simp.md

```diff
@@ -55,9 +55,43 @@ Using simp in the middle of proofs is a simp anti-pattern, which will produce br
 
 Conversely, if you ever manage to close a goal with simp, then take a look at the line before you ran simp. Could you have run simp one line earlier? How far back did simp start working? Even for goals where you didn't use simp at all -- could you have used simp for your last line? What about the last-but one? And so on.
 
+### Simp options.
+
+The behaviour of `simp` can be tweaked by simp variants and also by passing options to the algorithm. A good place to start is to look at the docstring for simp (write simp in VS Code and hover your mouse over it to see the docstring). Here are some examples, some of which are covered by the docstring and some of whichare not.
+
+1) `simp only [H1,H2,H3]` uses only lemmas `H1`, `H2`, and `H3` rather than `simp`s full collection of lemmas. Whyever might one want to do this in practice? Because sometimes simp simplifies things too much -- it might unfold things that you wanted to keep folded, for example. 
+
+2) `simp [- X]` stops `simp` from using lemma `X`. One could imagine using this as another solution when one finds `simp` doing more than you would like. Recall from above that `set_option trace.simplify.rewrite true` shows you exactly which lemmas `simp` is using.
+
+3) `simp * at *`. This simplifies everything in sight. Use if life is getting complicated.
+
+4) `simp {single_pass := tt}` -- this `single_pass` is a config option, one of around 16 at the time of writing. I guess it makes `simp` stop after it has managed to do one thing. As you can imagine, `simp only [h] single_pass := tt` here makes simp behave pretty much like `rw h`.
```

Comment:
The braces in `simp only [h] {single_pass := tt}` are not optional.

It doesn't stop `simp` after doing one thing, but it doesn't simp the RHS result of any performed simplifications. For example,
```
example (x : ℕ) : nat.gcd x x = x :=
by simp only [nat.gcd_def] {single_pass := tt}
```
does not go into an infinite loop of replacing `gcd` with itself when the single_pass option is true.